### DISCUSSION TITLE
Example uses for infinite sheds

### DISCUSSION THREAD

**Discussion by mikofski** [pvlib/pvlib-python#1432] [Ideas]
I can't wait until this module is available, and was thinking how can we showcases it's capabilities with lots of gallery examples?

1. infinite sheds can give hourly, daily, monthly, and annual beam shading from row-to-row (r2r), although this doesn't translate into electrical losses, if we assume linear, 70% electrical effect, or a constant shade impact factor we could summarize the effects of r2r on output - I think this would be a first for pvlib, right?
2. obviously BIFACIAL! duh! (head slap)
3. comparison of existing diffuse masking module vs. pvfactors vs. infinte sheds
    - total diffuse shade for bifacial (both sides)
    - total diffuse shade for monofacial (front side only)
4. develop a loss tree to compare with PVsyst and other software?

**Comment by cwhanse** [comment]
An example showing how to compute annual "bifacial gain", e.g., bifacial single-axis tracker energy / monofacial SAT energy, would be a good start.

**Comment by williamhobbs** [comment]
Has there been any more discussion on infinite_sheds gallery examples? I'm interested in one that uses the new diffuse fraction and shaded fraction outputs and @kandersolar's simple shade model for row-to-row self-shade losses. 

Here's a gist of (most of) what I had in mind, https://gist.github.com/williamhobbs/9d1fb23a64f1921400938250af6a67f0, resulting in one or more plots like this one:

![image](https://github.com/pvlib/pvlib-python/assets/45701090/420b3a71-f887-4609-a9a0-ff992566f594)


It could also be interesting to illustrate:
1. a linear model for thin film fixed and tracking (I already have those accounted for in the gist above, but don't actually run through it)
2. c-Si tracking w/ backtracking, using infinite_sheds to get diffuse shade losses
3. c-Si tracking w/ backtracking and _tracker clock drift_ to show power loss impacts of misaligned trackers

1 and 2 could be easy to add in the same example if more of the calculations were wrapped in functions, so they didn't have to be repeated 3-4 times. But then it might be harder to explain intermediate steps cleanly in the gallery?

**Reply by cwhanse** [reply]
I think what's in the gist tends to a more complex example, not in the code but in the question it addresses: how much power/energy is given up due to row-to-row shading? That requires two calculations +1 more for each shding model. Once the analysis is explained, and the calculations are presented, it will be a long-ish example. My reaction is to put the tracking/backtracking comparisons into other examples.

**Reply by williamhobbs** [reply]
Thanks for the feedback, @cwhanse. I realized it's easy to add a linear shade (think film) case to the gist example ([same gist link](https://gist.github.com/williamhobbs/9d1fb23a64f1921400938250af6a67f0)). What do you think about a cleaned-up and explained version of this as a standalone gallery example?

![image](https://github.com/pvlib/pvlib-python/assets/45701090/21c8e55c-e5ee-4841-952a-f0055b2d9928)

**Reply by cwhanse** [reply]
The harder part is writing the explanations :)

**Reply by williamhobbs** [reply]
Feedback on the updated gist would be great, https://gist.github.com/williamhobbs/9d1fb23a64f1921400938250af6a67f0. I tried copying Mark's illustration style from the `tracker_shaded_fraction` docs when explaining how I got creative with GCR for the top strings of a 2P system.

**Comment by williamhobbs** [comment]
@mikofski, how did you make this `pvlib.shading.tracker_shaded_fraction` illustration?

![image](https://github.com/pvlib/pvlib-python/assets/45701090/fcd225bd-d2c8-4dba-ab80-e32675291a22)

(https://github.com/mikofski/pvlib-python/blob/linear_shade_gh1690/docs/sphinx/source/_images/FSLR_irrad_shade_loss_slope_terrain.png)

It looks like it could have been PowerPoint (my drawing tool of choice, don't tell anyone 😬), but I figured I'd check in case there is some great PV geometry sketch tool I should know about...

**Reply by mikofski** [reply]
I admit it was pptx, but Kevin turned me on to asymptote, it’s a pretty nifty tool for sketches

**Reply by williamhobbs** [reply]
Thanks, I’ll have to try out asymptote. And nice to know I’m not the only one using PowerPoint for technical drawings.

**Comment by roshniagrawal1** [comment]
Hi @williamhobbs, 

Thank you for sharing the example to use infinite-shed model.

According to @mikofski's infinite shed model adopted in PVlib [[ref](https://mikofski.github.io/pvsc46_solarfarmer_bifacial/SolarFarmerBifacialModeling.pdf)],

**gcr = PV module height (H)/Row spacing (P)**

In the example code, you have used **pitch = pvrow_width/gcr** where pvrow_width = 4 meters. This seems to be wrong as Height of a module (H) should be used instead of width. Ofcourse, let's not confuse this with pvrow_height (h) which is 1 meters. The notations are from the reference for understanding.

Best,
Roshni

**Reply by mikofski** [reply]
Hi @roshniagrawal1 I think @williamhobbs [example gist](https://gist.github.com/williamhobbs/9d1fb23a64f1921400938250af6a67f0) is correct:

```python
# Site metadata
lat, lon = 32, -84
transposition_model = 'perez' 
shade_model = 'non_linear' # can be 'linear' (thin film) or 'non_linear' (c-Si)
mount = 'fixed_mount' # could be 'singleaxis_mount' or 'fixed_mount' 
module_orientation = '2P' # could be '1P' or '2P' for c-Si, ignored for thin film
n_cells_up = 12 # number of cells along the length of the modules

# row geometry info
gcr = 0.66
pvrow_height = 1
pvrow_width = 4
pitch = pvrow_width / gcr
```

The terms height and width are ambiguous and frequently interchanged. In this case, it’s a 2P orientation so the collector width refers to the length of active area perpendicular to the rows or 4m

**Reply by williamhobbs** [reply]
@roshniagrawal1, thanks for reviewing the example. @mikofski is correct about the terms being ambiguous. The example assumes that the modules are 2m x 1m and installed "2P", or 2-in-portrait, resulting in a 4m "wide" row. 

If you haven't seen it, we presented at PVPMC 2024 on modeling self-shade with an iteration of this approach. https://github.com/williamhobbs/2024_pvpmc_self_shade. And that was before the new shaded fraction function (https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.shading.shaded_fraction1d.html), so there are more ways to do this now.

**Reply by roshniagrawal1** [reply]
Thank you @mikofski and @williamhobbs for explaining this in detail! 

It has helped me gain clarity and understand the logic behind the calculations for difference configurations. The clarification about pvrow_width in the 2P setup makes a lot of sense now.

Thank you @williamhobbs once gain for sharing another example. I will check that out.
